dynamo/tools/clustering.py

```diff
@@ -227,31 +251,62 @@ def leiden(
-    "Louvain maximizes a modularity score for each community. The algorithm optimises the modularity in two elementary phases: (1) local moving of nodes; (2) aggregation of the network. In the local moving phase, individual nodes are moved to the community that yields the largest increase in the quality function. In the aggregation phase, an aggregate network is created based on the partition obtained in the local moving phase. Each community in this partition becomes a node in the aggregate network. The two phases are repeated until the quality function cannot be increased further." - cdlib
+    """Apply louvain clustering to adata.
+    For other community detection general parameters,
+    please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
+    "Louvain maximizes a modularity score for each community.
+    The algorithm optimises the modularity in two elementary phases:
+    (1) local moving of nodes;
+    (2) aggregation of the network.
+    In the local moving phase,
+    individual nodes are moved to the community that yields the largest increase in the quality function.
+    In the aggregation phase,
+    an aggregate network is created based on the partition obtained in the local moving phase.
+    Each community in this partition becomes a node in the aggregate network.
+    The two phases are repeated until the quality function cannot be increased further." - cdlib
 
-    Parameters
-    ----------
-    resolution : float, optional
-        change the size of the communities, default to 1.
-    randomize : bool, optional
-        "randomState instance or None, optional (default=None). If int, random_state is the seed used by the random number generator; If RandomState instance, random_state is the random number generator; If None, the random number generator is the RandomState instance used by np.random." - cdlib
+    Args:
+        adata: an adata object
+        resolution: determines clustering level of detail.
+            Please note that in louvain-igraph, increasing the parameter creates fewer clusters.
+            In our code, the resolution parameter in louvain is inverted to match the effect of leiden,
+            for instance, increasing resolution creates more clusters and decreasing it generates fewer.
+        use_weight: whether to use the weight of the edges in the clustering process. Default True
+        adj_matrix: the adjacency matrix to use for clustering. Default None
+        adj_matrix_key: adj_matrix_key in adata.obsp used for clustering. Default None
+        randomize: randomState instance or None, optional (default=None).
+            If int, random_state is the seed used by the random number generator;
+            If RandomState instance, random_state is the random number generator;
+            If None, the random number generator is the RandomState instance used by np.random." - louvain-igraph
```

Comment:
don't be lazy with - louvain-igraph. say something like, `according to louvain-igraph's louvain algorithm's documentation`

Please don't just copy my phrase. make it precise

Reply:
done